flake.nix

```diff
@@ -0,0 +1,83 @@
+{
+  description = "A library for probabilistic programming in Haskell.";
+
+  nixConfig = {
+    extra-substituters = [
+      "https://tweag-monad-bayes.cachix.org"
+    ];
+    extra-trusted-public-keys = [
+      "tweag-monad-bayes.cachix.org-1:tmmTZ+WvtUMpYWD4LAkfSuNKqSuJyL3N8ZVm/qYtqdc="
+    ];
+  };
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-unstable";
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
```

Comment:
Can we clean this up? i.e. 

```
flake-compat = { 
  url = ...;
  flake = ...;
}
```

The duplication hurts my eyes.

Reply:
You choice :)